crates/router/src/types/api/admin.rs

```diff
@@ -80,169 +83,189 @@ impl ForeignTryFrom<domain::MerchantAccount> for MerchantAccountResponse {
-                .transpose()
+    let payment_link_config_value = request
+        .payment_link_config
+        .map(|pl_config| {
+            pl_config
+                .encode_to_value()
                 .change_context(errors::ApiErrorResponse::InvalidDataValue {
-                    field_name: "authentication_connector_details",
-                })?,
-            payout_link_config: request
-                .payout_link_config
-                .as_ref()
-                .map(Encode::encode_to_value)
-                .transpose()
-                .change_context(errors::ApiErrorResponse::InvalidDataValue {
-                    field_name: "payout_link_config",
-                })?,
-            is_connector_agnostic_mit_enabled: request.is_connector_agnostic_mit_enabled,
-            is_extended_card_info_enabled: None,
-            extended_card_info_config: None,
-            use_billing_as_payment_method_billing: request
-                .use_billing_as_payment_method_billing
-                .or(Some(true)),
-            collect_shipping_details_from_wallet_connector: request
-                .collect_shipping_details_from_wallet_connector
-                .or(Some(false)),
-            collect_billing_details_from_wallet_connector: request
-                .collect_billing_details_from_wallet_connector
-                .or(Some(false)),
+                    field_name: "payment_link_config_value",
+                })
         })
-    }
+        .transpose()?;
+    let outgoing_webhook_custom_http_headers = request
+        .outgoing_webhook_custom_http_headers
+        .async_map(|headers| create_encrypted_data(key_store, headers))
+        .await
+        .transpose()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Unable to encrypt outgoing webhook custom HTTP headers")?;
+
+    Ok(storage::business_profile::BusinessProfileNew {
+        profile_id,
+        merchant_id: merchant_account.merchant_id,
+        profile_name: request.profile_name.unwrap_or("default".to_string()),
+        created_at: current_time,
+        modified_at: current_time,
+        return_url: request
+            .return_url
+            .map(|return_url| return_url.to_string())
+            .or(merchant_account.return_url),
+        enable_payment_response_hash: request
+            .enable_payment_response_hash
+            .unwrap_or(merchant_account.enable_payment_response_hash),
+        payment_response_hash_key: Some(payment_response_hash_key),
+        redirect_to_merchant_with_http_post: request
+            .redirect_to_merchant_with_http_post
+            .unwrap_or(merchant_account.redirect_to_merchant_with_http_post),
+        webhook_details: webhook_details.or(merchant_account.webhook_details),
+        metadata: request.metadata,
+        routing_algorithm: Some(serde_json::json!({
+            "algorithm_id": null,
+            "timestamp": 0
+        })),
```

Comment:
Do we still need this hardcoded value?

Reply:
@SanchithHegde this was hardcoded for backward compatibility. 
cc: @Aprabhat19